### ISSUE TITLE
WebSocketSubject cannot be retried.

### ISSUE THREAD

**Issue by blesh** [ReactiveX/rxjs#1466]
**RxJS version:** 5.0.0-beta.2

**Code to reproduce:**

```js
let socket = new WebSocketSubject('ws://badurlhere');

socket.retryWhen((errs) => errs.switchMap(() => Observable.timer(1000))
  .subscribe(::console.log);
```

**Expected behavior:**

connection fails and then retries once a second.

**Actual behavior:**

connection fails and then `UnsubscriptionError` is thrown, subsequent subscriptions do not work

**Additional information:**

This apparently has to do with some changes to Subject and unsubscription error handling. I'm not sure though. We should definitely add some tests around this scenario, because it's important. WebSocketSubject was supposed to be reusable.

I found this will trying out the `multiplex` method in production for the first time. :\ Bummer.

**Comment by skaapgif** [comment]
I'm going to have a stab at this, I think I've identified the cause, but I'll write a failing test case to isolate it.

**Comment by skaapgif** [comment]
I created a failing test case and a fix:
https://github.com/skaapgif/rxjs/commit/9fd31cc0a0fdb91fc5152c376995918a57689057

When I apply the fix to my own test environment it solves the problem, but the test still fails for some reason. Any ideas?

The problem is each time a multiplex subject unsubscribes, it sends the `unsub` message to announce this to the server. If, however, the underlying socket closes, the multiplex subject still tries to send it's `unsub` message over an already unsubscribed subject which throws an `UnsubscriptionError`.

The current fix check's the underlying socket's ready state with `self.socket.readyState < 3` before sending an `unsub` message, but it might also be possible to check `self.isUnsubscribed`.

**Comment by fs123** [comment]
It looks, that the bug does no longer exists since beta 8. See added test in PR 1737 above.

**Comment by fs123** [comment]
It looks, that the bug does no longer exists since beta 8. See added test in PR 1738 above.

**Comment by blesh** [comment]
> I created a failing test case and a fix: skaapgif@9fd31cc

@skaapgif did you already PR your fix and test for the unsub message?

**Comment by blesh** [comment]
Closed by #1738

**Comment by skaapgif** [comment]
I was still awaiting feedback, but @fs123's test is very similar and more thorough. I was able to reproduce the error without calling `retry(1)` though.
